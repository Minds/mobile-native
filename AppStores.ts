import { createLegacyStores, TLegacyStores } from './src/common/contexts';

let stores: any;

export const getStores = function (): TLegacyStores {
  if (!stores) {
    stores = createLegacyStores();
  }
  return stores;
};
