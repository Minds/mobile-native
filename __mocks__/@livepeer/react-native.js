// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
