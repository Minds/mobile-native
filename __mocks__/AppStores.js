export default {
  __esModule: true,
  getStores: jest.fn(),
};
