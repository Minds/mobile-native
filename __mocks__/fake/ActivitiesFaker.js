function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
