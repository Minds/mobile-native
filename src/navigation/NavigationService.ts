//@ts-nocheck
import {
  CommonActions,
  StackActions,
  SwitchActions,
} from '@react-navigation/native';

let _navigator = null;

function getStateFrom(nav) {
  if (nav.routes && nav.routes[nav.index].state) {
    return getStateFrom(nav.routes[nav.index].state);
  }
  return nav.routes[nav.index];
}

export function setTopLevelNavigator(navigatorRef) {
  _navigator = navigatorRef;
}

function getCurrentState() {
  const root = _navigator.getRootState();
  return getStateFrom(root);
}

function navigate(...args) {
  _navigator.navigate(...args);
}

function push(...args) {
  _navigator.dispatch(StackActions.push(...args));
}

function goBack() {
  _navigator.dispatch(CommonActions.goBack());
}

function jumpTo(route) {
  _navigator.dispatch(SwitchActions.jumpTo({ route }));
}

function reset(routeName, params) {
  const resetAction = StackActions.reset({
    index: 0,
    actions: [CommonActions.navigate({ routeName: routeName })],
  });
  _navigator.dispatch(resetAction);
}

function addListener(name, fn) {
  return _navigator?.addListener(name, fn);
}

// add other navigation functions that you need and export them

export default {
  navigate,
  jumpTo,
  reset,
  getCurrentState,
  push,
  setTopLevelNavigator,
  goBack,
  addListener,
};
