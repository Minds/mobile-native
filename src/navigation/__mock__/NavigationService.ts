export default {
  navigate: jest.fn(),
  jumpTo: jest.fn(),
  reset: jest.fn(),
  getCurrentState: jest.fn(),
  push: jest.fn(),
  setTopLevelNavigator: jest.fn(),
  goBack: jest.fn(),
  addListener: jest.fn(),
};
