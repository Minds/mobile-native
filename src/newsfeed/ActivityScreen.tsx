import React, { useEffect } from 'react';
import { AppStackParamList } from '../navigation/NavigationTypes';
import { RouteProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { useLocalStore, observer } from 'mobx-react';
import ActivityFullScreen from '../discovery/v2/viewer/ActivityFullScreen';
import SingleEntityStore from '../common/stores/SingleEntityStore';
import ActivityModel from './ActivityModel';
import { FLAG_VIEW } from '../common/Permissions';
import OffsetFeedListStore from '../common/stores/OffsetFeedListStore';
import CenteredLoading from '../common/components/CenteredLoading';
import type BlogModel from '../blogs/BlogModel';

export type ActivityRouteProp = RouteProp<AppStackParamList, 'Activity'>;
type ActivityNavigationProp = StackNavigationProp<
  AppStackParamList,
  'Activity'
>;

type PropsType = {
  route: ActivityRouteProp;
  navigation: ActivityNavigationProp;
};

/**
 * Single activity screen
 */
const ActivityScreen = observer((props: PropsType) => {
  const store = useLocalStore(
    (p: PropsType) => ({
      loading: false,
      entityStore: new SingleEntityStore<ActivityModel>(),
      async loadEntity() {
        const params = p.route.params;
        if (
          params.entity &&
          (params.entity.guid || params.entity.entity_guid)
        ) {
          const urn =
            'urn:activity:' + (params.entity.guid || params.entity.entity_guid);

          const entity = ActivityModel.checkOrCreate(params.entity);

          if (!entity.can(FLAG_VIEW, true)) {
            props.navigation.goBack();
            return;
          }

          store.entityStore.loadEntity(urn, entity, true);

          // change metadata source
          if (params.entity._list && params.entity._list.metadataService) {
            params.entity._list.metadataService.pushSource('single');
          }
        } else {
          const urn = 'urn:activity:' + params.guid;
          await store.entityStore.loadEntity(urn, undefined, false);

          if (
            !store.entityStore.entity ||
            !store.entityStore.entity.can(FLAG_VIEW, true)
          ) {
            props.navigation.goBack();
            return;
          }

          // in case it is opened from a deeplink and it is a blog we should replace withs blog screen
          if (store.entityStore.entity.subtype === 'blog') {
            props.navigation.replace('BlogView', {
              blog: store.entityStore.entity as BlogModel,
            });
          }
        }

        if (params.entity && params.entity._list) {
          // this second condition it's for legacy boost feed
          if (params.entity._list instanceof OffsetFeedListStore) {
            params.entity._list.addViewed(params.entity);
          } else {
            params.entity._list.viewed.addViewed(
              params.entity,
              params.entity._list.metadataService,
            );
          }
        }
      },
    }),
    props,
  );

  useEffect(() => {
    store.loadEntity();
  }, [store]);

  if (!store.entityStore.entity) {
    return <CenteredLoading />;
  }

  return <ActivityFullScreen entity={store.entityStore.entity} />;
});

export default ActivityScreen;
