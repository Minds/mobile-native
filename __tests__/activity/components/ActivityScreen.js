import 'react-native';
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import ActivityScreen from '../../../src/newsfeed/ActivityScreen';
import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';
import entitiesService from '../../../src/common/services/entities.service';
import ActivityModel from '../../../src/newsfeed/ActivityModel';
import NavigationService from '../../../src/navigation/NavigationService';
import {
  useNavigation,
  useRoute,
} from '../../../__mocks__/@react-navigation/native';
import { getStores } from '../../../AppStores';

getStores.mockReturnValue({
  newsfeed: {},
});

jest.mock('react-native-system-setting', () => {
  return {
    getVolume: jest.fn(() => Promise.resolve())
  }
})

jest.mock('react-native-safe-area-context');
jest.mock('react-native-material-menu');
jest.mock('react-native-silent-switch');
jest.mock('@react-navigation/native');
jest.mock('react-native-gesture-handler');
jest.mock('react-native-redash');
jest.mock('../../../src/common/components/KeyboardSpacingView');
jest.mock('reanimated-bottom-sheet', () => 'BottomSheet');
jest.mock('@gorhom/bottom-sheet', () => {
  const react = require('react-native');

  return {
    BottomSheetFlatList: react.FlatList,
    TouchableOpacity: react.TouchableOpacity,
  };
});

jest.mock('../../../src/newsfeed/NewsfeedService');
jest.mock('../../../src/newsfeed/activity/Activity', () => 'Activity');
jest.mock(
  '../../../src/common/components/CenteredLoading',
  () => 'CenteredLoading',
);
// jest.mock('../../../src/comments/CommentsStore');
// jest.mock('../../../src/comments/CommentsStoreProvider');
jest.mock('../../../src/common/services/entities.service');
jest.mock('../../../src/navigation/NavigationService');
jest.mock('../../../src/common/hooks/use-stores');
jest.mock(
  '../../../src/comments/v2/CommentBottomSheet',
  () => 'CommentBottomSheet',
);

// react testing library has a problem with the fake timers
// https://github.com/callstack/react-native-testing-library/issues/391
jest.useRealTimers();

describe('Activity screen component', () => {
  it('renders correctly ', async (done) => {
    try {
      const navigation = {
        push: jest.fn(),
        goBack: jest.fn(),
      };
      useNavigation.mockReturnValue(navigation);
      useRoute.mockReturnValue({ params: {} });
      NavigationService.getCurrentState.mockReturnValue({ params: {} });

      const route = {
        routeName: 'some',
        params: { entity: activitiesServiceFaker().load(1).activities[0] },
      };

      entitiesService.single.mockResolvedValue(
        ActivityModel.create(route.params.entity),
      );

      const { toJSON, getByA11yLabel } = render(
        <ActivityScreen navigation={navigation} route={route} />,
      );

      await waitFor(() => getByA11yLabel('touchableTextCopy'), {
        timeout: 4000,
        interval: 100,
      });

      // should show the activity
      await expect(toJSON()).toMatchSnapshot();

      done();
    } catch (e) {
      done.fail(e);
    }
  });
});
