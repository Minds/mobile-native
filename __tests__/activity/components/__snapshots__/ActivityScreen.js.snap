// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity screen component renders correctly  1`] = `
<View
  style={
    Array [
      Object {
        "fontScale": 2,
        "height": 1334,
        "scale": 2,
        "width": 750,
      },
      undefined,
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
          },
          undefined,
          Array [
            undefined,
            Object {
              "borderBottomWidth": 0,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            },
            Object {
              "paddingTop": -10,
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 13,
          }
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 45,
              },
              Array [
                Object {
                  "elevation": 4,
                  "left": 0,
                  "position": "absolute",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 3,
                },
                undefined,
                Object {
                  "top": 5,
                },
                Array [
                  undefined,
                  Object {
                    "marginLeft": -17,
                    "marginRight": -5,
                    "paddingTop": 5,
                    "position": undefined,
                    "top": undefined,
                    "width": 50,
                  },
                ],
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰅁
        </Text>
        <View
          onPress={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18.5,
                  "height": 37,
                  "width": 37,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "headers": Object {
                    "App-Version": "3.8.0",
                    "Cache-Control": "no-cache, no-store, must-revalidate",
                    "Content-Type": "application/json",
                    "Pragma": "no-cache",
                  },
                  "uri": "https://cdn.minds.com/icon/824853017709780997/medium/undefined",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "marginLeft": 10,
              "paddingRight": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <View
              onPress={[Function]}
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#444",
                      "fontFamily": "Roboto",
                      "fontSize": 17,
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                      },
                    ],
                  ]
                }
              >
                <Text
                  onPress={[Function]}
                  style={
                    Array [
                      Array [
                        Object {
                          "backgroundColor": "#000",
                          "bottom": 0,
                          "left": 0,
                          "opacity": 0.4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                      ],
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#242A30",
                      "flex": 1,
                      "height": 45.5,
                      "transform": Array [
                        Object {
                          "translateY": 45.5,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "height": 40,
                          "justifyContent": "center",
                        },
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#242A30",
                          "height": 40,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#757575",
                            "fontSize": 14,
                          },
                        ]
                      }
                    >
                      Actions
                    </Text>
                  </View>
                  <RCTScrollView
                    scrollEnabled={false}
                  >
                    <View />
                  </RCTScrollView>
                  <View
                    activeOpacity={1}
                    onPress={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "height": 50,
                          "justifyContent": "center",
                          "marginTop": 6,
                        },
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#1A2025",
                          "height": 50,
                          "justifyContent": "center",
                          "marginTop": 6,
                        },
                      ]
                    }
                    underlayColor="#F4F4F4"
                  >
                    <Text
                      style={
                        Array [
                          Array [
                            Object {
                              "fontSize": 18,
                            },
                          ],
                          Object {
                            "color": "#007AFF",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Array [
          null,
          undefined,
          Object {
            "flexGrow": 1,
            "justifyContent": "center",
          },
        ]
      }
      onContentSizeChange={[Function]}
      onLayout={[Function]}
    >
      <View>
        <View
          accessibilityLabel="touchableTextCopy"
          onLongPress={[Function]}
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                null,
              ]
            }
          >
            <Text
              selectable={true}
              style={
                Array [
                  Array [
                    undefined,
                    undefined,
                  ],
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              TITLE
            </Text>
            <Text
              selectable={false}
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              Message
            </Text>
          </View>
          <View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              onRequestClose={[Function]}
              scrollHorizontal={false}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                forwardedRef={[Function]}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#1E242A",
                    "bottom": 0,
                    "height": 1334,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 750,
                  }
                }
              />
              <View
                forwardedRef={[Function]}
                hideModalContentWhileAnimating={false}
                onModalHide={[Function]}
                onModalWillHide={[Function]}
                onModalWillShow={[Function]}
                pointerEvents="box-none"
                scrollHorizontal={false}
                scrollOffset={0}
                scrollOffsetMax={0}
                scrollTo={null}
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 37.5,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
                supportedOrientations={
                  Array [
                    "portrait",
                    "landscape",
                  ]
                }
                swipeThreshold={100}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#242424",
                        "marginBottom": 30,
                      }
                    }
                  >
                    
                  </Text>
                  <View
                    style={
                      Array [
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "marginBottom": 30,
                        },
                      ]
                    }
                  >
                    <RCTScrollView
                      data={null}
                      disableVirtualization={false}
                      extraData=""
                      getItem={[Function]}
                      getItemCount={[Function]}
                      horizontal={false}
                      initialNumToRender={10}
                      keyExtractor={[Function]}
                      maxToRenderPerBatch={10}
                      onContentSizeChange={[Function]}
                      onEndReachedThreshold={2}
                      onLayout={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      onScrollToIndexFailed={[Function]}
                      removeClippedSubviews={false}
                      renderItem={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      updateCellsBatchingPeriod={50}
                      viewabilityConfigCallbackPairs={Array []}
                      windowSize={21}
                    >
                      <View />
                    </RCTScrollView>
                  </View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderRadius": 28,
                        "height": 55,
                        "margin": 7,
                        "overflow": "hidden",
                        "shadowColor": "rgba(0,0,0, .4)",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": 1,
                        },
                        "shadowOpacity": 1,
                        "shadowRadius": 1,
                        "width": 55,
                        "zIndex": 1000,
                      }
                    }
                  >
                    <View
                      activeOpacity={0.3}
                      disabled={false}
                      onPress={[Function]}
                      underlayColor="transparent"
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "borderRadius": 28,
                            "height": 52,
                            "justifyContent": "center",
                            "width": 52,
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "black",
                                "fontSize": 24,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            
            Mar 25th, 2018
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            1712165,
            4279902245,
          ],
          "endPoint": undefined,
          "locations": undefined,
          "startPoint": undefined,
        }
      }
      style={
        Object {
          "bottom": 0,
          "height": 40,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingBottom": -10,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderTopWidth": 0.5,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "paddingHorizontal": 10,
            },
            undefined,
          ]
        }
      >
        <View
          onPress={[Function]}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
          testID="Thumb up activity button"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 21,
                },
                Array [
                  undefined,
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          onPress={[Function]}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
          testID="Thumb down activity button"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 21,
                },
                Array [
                  undefined,
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          onPress={[Function]}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
          testID=""
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 21,
                },
                Array [
                  undefined,
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              }
            }
          >
            <Text
              size={15}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Menu
          button={
            <preventDoubleTap(TouchableOpacity)
              onPress={[Function]}
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
              testID="Remind activity button"
            >
              <Icon
                allowFontScaling={false}
                name="repeat"
                size={21}
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              />
              <Counter
                count={0}
                size={15}
              />
            </preventDoubleTap(TouchableOpacity)>
          }
        >
          <MenuItem
            onPress={[Function]}
            textStyle={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  Array [
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
              Remind
          </MenuItem>
          <MenuItem
            onPress={[Function]}
            textStyle={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  Array [
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
              Quote
          </MenuItem>
        </Menu>
      </View>
    </View>
  </View>
  <CommentBottomSheet
    commentsStore={
      CommentsStore {
        "attachment": AttachmentStore {
          "cancelOrDelete": [Function],
          "fileName": null,
          "guid": "",
          "transcoding": false,
        },
        "comments": Array [],
        "embed": RichEmbedStore {
          "_richEmbedFetchTimer": null,
          "clearRichEmbedAction": [Function],
          "richEmbedUrl": "",
          "setRichEmbedPromise": null,
        },
        "entity": Object {
          "allow_comments": true,
          "attachment_guid": false,
          "blurb": false,
          "comments:count": undefined,
          "containerObj": undefined,
          "container_guid": "activityguid0",
          "custom_data": false,
          "custom_type": false,
          "description": "Congratulations! ",
          "edited": "",
          "entity_guid": "",
          "getLockType": [Function],
          "guid": "activityguid0",
          "impressions": 0,
          "isOwner": [Function],
          "is_visible": false,
          "mature": false,
          "mature_visibility": false,
          "message": "Message",
          "ownerObj": Object {
            "__list": null,
            "allow_comments": true,
            "blocked": undefined,
            "canary": false,
            "confirmEmail": [Function],
            "disable_autoplay_videos": undefined,
            "email_confirmed": false,
            "eth_wallet": "",
            "founder": false,
            "groupsCount": 0,
            "guid": "824853017709780997",
            "impressions": undefined,
            "isOwner": [Function],
            "is_admin": false,
            "last_accepted_tos": 0,
            "mature": false,
            "mature_visibility": false,
            "mode": 0,
            "nsfw": Array [],
            "onchain_booster": 0,
            "pending_subscribe": false,
            "permissions": Object {},
            "plus": false,
            "pro": false,
            "rewards": false,
            "subscribed": undefined,
            "subscribers_count": undefined,
            "subscriptions_count": 0,
            "subtype": false,
            "tags": Array [],
            "time_created": "1522036284",
            "type": "user",
            "urn": "",
            "username": "",
            "verified": false,
            "wire_rewards": undefined,
          },
          "owner_guid": "824853017709780997",
          "parent_guid": "838106762591510528",
          "paywall": "",
          "paywall_unlocked": false,
          "perma_url": false,
          "permissions": Object {},
          "pinned": false,
          "reminds": 0,
          "rowKey": "something0",
          "shouldBeBlured": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "subtype": "",
          "thumbnail_src": false,
          "thumbs:down:count": undefined,
          "thumbs:down:user_guids": undefined,
          "thumbs:up:count": undefined,
          "thumbs:up:user_guids": undefined,
          "time_created": "1522036284",
          "title": "TITLE",
          "type": "activity",
          "urn": "",
          "username": "",
          "wire_totals": Object {
            "tokens": 1000000000000000000,
          },
        },
        "errorLoadingNext": false,
        "errorLoadingPrevious": false,
        "focusedUrn": null,
        "guid": "",
        "level": 0,
        "loadNext": "",
        "loadPrevious": "",
        "loaded": false,
        "loadingNext": false,
        "loadingPrevious": false,
        "mature": 0,
        "onAttachedMedia": [Function],
        "parent": null,
        "post": [Function],
        "refreshing": false,
        "reversed": true,
        "saving": false,
        "selectMediaType": [Function],
        "showInput": false,
        "socketRoomName": "",
        "text": "",
      }
    }
    hideContent={false}
  />
</View>
`;
